aws/components/directory/setup.ftl

```diff
@@ -0,0 +1,254 @@
+
+            [#if !linkTarget?has_content]
+                [#continue]
+            [/#if]
+
+            [#local linkTargetCore = linkTarget.Core ]
+            [#local linkTargetConfiguration = linkTarget.Configuration ]
+            [#local linkTargetResources = linkTarget.State.Resources ]
+            [#local linkTargetAttributes = linkTarget.State.Attributes ]
+
+            [#if deploymentSubsetRequired(DIRECTORY_COMPONENT_TYPE, true)]
+                [@createSecurityGroupRulesFromLink
+                    occurrence=occurrence
+                    groupId=cacheSecurityGroupId
+                    linkTarget=linkTarget
+                    inboundPorts=[ "ssh" ]
+                /]
+            [/#if]
+
+        [/#if]
+    [/#list]
+
+    [#-- Secret Management --]
+    [#local secretStoreLink = getLinkTarget(occurrence, solution.RootCredentials.SecretStore) ]
+    [#local passwordSecretKey = "password" ]
+
+    [#if secretStoreLink?has_content ]
+
+        [@setupComponentSecret
+            occurrence=occurrence
+            secretStoreLink=secretStoreLink
+            kmsKeyId=cmkKeyId
+            secretComponentResources=resources["rootCredentials"]
+            secretComponentConfiguration=
+                solution.RootCredentials.Secret + {
+                    "Generated" : {
+                        "Content" : { "username" : solution.RootCredentials.Username },
+                        "SecretKey" : passwordSecretKey
+                    }
+                }
+            componentType=DIRECTORY_COMPONENT_TYPE
+        /]
+    [#else]
+        [@fatal
+            message="Could not find link to secret store or link was invalid"
+            detail="Add a link to a secret store component which will manage the root credentials"
+            context=solution.RootCredentials.SecretStore
+        /]
+    [/#if]
+
+    [#-- Output Generation --]
+    [#if deploymentSubsetRequired(DIRECTORY_COMPONENT_TYPE, true)]
+
+        [#-- Network Security --]
+        [@createSecurityGroup
+            id=securityGroupId
+            name=securityGroupName
+            vpcId=vpcId
+            occurrence=occurrence
+        /]
+
+        [@createSecurityGroupRulesFromNetworkProfile
+            occurrence=occurrence
+            groupId=securityGroupId
+            networkProfile=networkProfile
+            inboundPorts=brokerPorts
+        /]
+
+        [#local ingressNetworkRule = {
+                "Ports" : brokerPorts,
+                "IPAddressGroups" : solution.IPAddressGroups
+        }]
+
+        [@createSecurityGroupIngressFromNetworkRule
+            occurrence=occurrence
+            groupId=securityGroupId
+            networkRule=ingressNetworkRule
+        /]
+
+        [#if !hibernate]
+
+            [#if ! testMaintenanceWindow(solution.MaintenanceWindow)]
+                [@fatal message="Maintenance window incorrectly configured" context=solution /]
+                [#return]
+            [/#if]
```

Comment:
Does the AD service have maintenance windows?

Reply:
No, Missed it why slicing and dicing